bloptools/bayesian/__init__.py

```diff
@@ -288,13 +247,153 @@ def tell(self, new_table=None, append=True, train=True, **kwargs):
             try:
                 self.train_models()
             except botorch.exceptions.errors.ModelFittingError:
-                if self._initialized:
+                if self.initialized:
                     self._set_hypers(cached_hypers)
                 else:
                     raise RuntimeError("Could not fit model on initialization!")
 
         self.constraint = GenericDeterministicModel(f=lambda x: self.classifier.probabilities(x)[..., -1].squeeze(-1))
 
+    def ask(self, acq_func_identifier="qei", n=1, route=True, sequential=True, **acq_func_kwargs):
```

Comment:
I am thinking about the `acq_func_identifier` values: maybe we can define an Enum class with the values and access it via `<Class>.qei`? Maybe even the dataclass with the fields to match the ones from the config file?

That way we (1) can easily get the parameters programmatically, (2) can have it in the docs, and (3) can avoid mistyped values by validating input before passing the input to `get_acquisition_function`.

Reply:
This is a good idea, but I don't quite understand what you mean by using Enum here. Let's chat in person and work it out.

Reply:
Discussed, we'll consider it after the typing annotations are implemented.